sensor/common/scannerclient/singleton.go

```diff
@@ -1,35 +1,55 @@
 package scannerclient
 
 import (
+	"github.com/stackrox/rox/pkg/centralsensor"
 	"github.com/stackrox/rox/pkg/env"
 	"github.com/stackrox/rox/pkg/features"
 	"github.com/stackrox/rox/pkg/sync"
 	"github.com/stackrox/rox/pkg/utils"
+	"github.com/stackrox/rox/sensor/common/centralcaps"
 )
 
 var (
-	once          sync.Once
-	scannerClient ScannerClient
+	scannerClient      ScannerClient
+	scannerClientMutex sync.Mutex
 
 	isScannerV4Enabled = features.ScannerV4.Enabled()
 )
 
 // GRPCClientSingleton returns a gRPC ScannerClient to a local Scanner.
 // Only one ScannerClient per Sensor is required.
 func GRPCClientSingleton() ScannerClient {
```

Comment:
just realized how dangerous this could be if we don't call this every time (dangerous as in, if the caller just calls it once and never tries again, then it may never get the new version). Seems like we do call this every time, though, so that's good

Reply:
Yes indeed, in the past would have been OK because Central would have restarted if lost connection - now however need this reset or things will get stale.